consensus/misc/eip1559.go

```diff
@@ -107,8 +107,71 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseF
 		return big.NewInt(10000000) // 0.01 Gwei
 	}
 
-	scalar, overhead := ReadL2BaseFeeCoefficients()
-	return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	if parent == nil || parent.Number == nil || !config.IsFeynman(parent.Time) {
+		scalar, overhead := ReadL2BaseFeeCoefficients()
+		return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	}
+	return calcBaseFeeFeynman(config, parent)
+}
+
+// calcBaseFeeFeynman calculates the basefee of the header for Feynman fork.
+func calcBaseFeeFeynman(config *params.ChainConfig, parent *types.Header) *big.Int {
+	baseFeeEIP1559 := calcBaseFeeEIP1559(config, parent)
+	baseFee := new(big.Int).Set(baseFeeEIP1559)
+	baseFee.Add(baseFee, config.ProvingBaseFee)
+
+	return baseFee
+}
+
+// CalcBaseFee calculates the basefee of the header.
+func calcBaseFeeEIP1559(config *params.ChainConfig, parent *types.Header) *big.Int {
```

Comment:
this should have extensive unit test and documentation about what are max/min values and how the fee adjusts.